prediction_market_agent/agents/arbitrage_agent/deploy.py

```diff
@@ -0,0 +1,193 @@
+        correlated_markets = []
+        related = self.pinecone_handler.find_nearest_questions_with_threshold(
+            limit=10, text=market.question
+        )
+        omen_markets = self.subgraph_handler.get_omen_binary_markets(
+            limit=len(related),
+            id_in=[i.market_address.lower() for i in related],
+            resolved=False,
+        )
+
+        # Note that negative correlation is hard - e.g. for the US presidential election, markets on each candidate are not seen as -100% correlated.
+        for related_market in omen_markets:
+            result = self.chain.invoke(
+                {
+                    "main_market_question": market,
+                    "related_market_question": related_market,
+                },
+                config=get_langfuse_langchain_config(),
+            )
+            if related_market.id != market.id and self.is_correlation_above_threshold(
+                result.correlation
+            ):
+                related_agent_market = OmenAgentMarket.from_data_model(related_market)
+                correlated_markets.append(
+                    CorrelatedMarketPair(
+                        main_market=market,
+                        related_market=related_agent_market,
+                        correlation=result.correlation,
+                    )
+                )
+        return correlated_markets
+
+    @observe()
+    def build_trades_for_correlated_markets(
+        self, pair: CorrelatedMarketPair
+    ) -> list[Trade]:
+        market_to_bet_yes, market_to_bet_no = pair.main_market, pair.related_market
+        total_amount: BetAmount = pair.main_market.get_tiny_bet_amount()
+        # Split between main_market and related_market
+        amount_yes, amount_no = pair.split_bet_amount_between_yes_and_no(
+            total_amount.amount
+        )
+        return [
+            Trade(
+                trade_type=TradeType.BUY,
+                outcome=True,
+                amount=TokenAmount(
+                    amount=amount_yes, currency=market_to_bet_yes.currency
+                ),
+            ),
+            Trade(
+                trade_type=TradeType.BUY,
+                outcome=False,
+                amount=TokenAmount(
+                    amount=amount_no, currency=market_to_bet_no.currency
+                ),
+            ),
+        ]
+
+    @observe()
+    def build_trades(
+        self,
+        market: AgentMarket,
+        answer: ProbabilisticAnswer,
+        existing_position: Position | None,
+    ) -> list[Trade]:
+        trades = []
+        correlated_markets = self.get_correlated_markets(market=market)
+        # For each correlated market, we want to place YES/NO or NO/YES trades.
+        for pair in correlated_markets:
+            if pair.potential_profit_per_bet_unit > 0:
+                trades_for_pair = self.build_trades_for_correlated_markets(pair)
+                trades.extend(trades_for_pair)
+
+        return trades
```

Comment:
I found it easier to build out the logic inside `build_trades` than to implement a whole new strategy, which would not follow the current interface on `BettingStrategy`.